### ISSUE TITLE
Polymorhpic: Type error: Argument 3 passed to Silber\Bouncer\Clipboard::Silber\Bouncer\{closure}() must be of the type array, none given

### ISSUE THREAD

**Issue by micfai** [JosephSilber/bouncer#73]
I am trying out the polymorphic branch and get this error when using

```php
Bouncer::allows( $permission );
```

Issue #59 is the same but it doesn't appear the fix got put into this branch.

I fixed it manually in my local install so I can continue to test it out.

**Comment by Keoghan** [comment]
I've also just started testing the polymorphic branch and ran into this issue (Laravel 5.1).

The fix was a little different to last time, but this appears to solve it. In Clipboard.php at line 30.

From:
```php
$gate->before(function ($authority, $ability, array $arguments) {
            if (count($arguments) > 1) {
                return;
            }
```

To:
```php
$gate->before(function ($authority, $ability, $arguments = []) {
            $arguments = (array) $arguments;
            if (count($arguments) > 1) {
                return;
            }
```

**Comment by micfai** [comment]
The closure arguments are different in this branch but the actual fix itself to set a default on the $arguments arguments is the same.

**Comment by Keoghan** [comment]
Unfortunately just setting the default on arguments didn't work for me, as when using a call to 

```php
$gate->denies('permission', $auth->user())
```

I received errors regarding the third param not being an array. 

```ErrorException in Clipboard.php line 30:
Argument 3 passed to Silber\Bouncer\Clipboard::Silber\Bouncer\{closure}() must be of the type array, object given```

Then when simple removing the type hint, a break was created elsewhere ```
FatalErrorException in Clipboard.php line 147:
Class name must be a valid object or a string
```
- until I force $arguments to be an a array, after which everything appears to run as intended.

**Comment by JosephSilber** [comment]
I think I'm jsut going to merge the polymorphic branch into master, and tag a 1.0 alpha.

**Comment by Keoghan** [comment]
Happy to give it a run if you do :)

**Comment by micfai** [comment]
I would be fine with you doing that.

**Comment by JosephSilber** [comment]
Here you go: https://github.com/JosephSilber/bouncer/releases/tag/v1.0.0-alpha.1

@micfai please try the new 1.0 alpha. If this bug has been resolved, kindly close this issue.

**Comment by micfai** [comment]
It appears to be resolved with v1.0.0-alpha.1
